mongodbatlas/fw_data_source_mongodbatlas_alert_configurations.go

```diff
@@ -0,0 +1,225 @@
+package mongodbatlas
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/listvalidator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	matlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+const alertConfigurationsDataSourceName = "alert_configurations"
+
+var _ datasource.DataSource = &AlertConfigurationDS{}
+var _ datasource.DataSourceWithConfigure = &AlertConfigurationDS{}
+
+type tfAlertConfigurationsDSModel struct {
+	ID          types.String                  `tfsdk:"id"`
+	ProjectID   types.String                  `tfsdk:"project_id"`
+	OutputType  []string                      `tfsdk:"output_type"`
+	ListOptions []tfListOptionsModel          `tfsdk:"list_options"`
+	Results     []tfAlertConfigurationDSModel `tfsdk:"results"`
```

Comment:
it may be worth using a pointer here since the struct can get quite big
```suggestion
	Results     []*tfAlertConfigurationDSModel `tfsdk:"results"`
```

Reply:
Not sure if I find value in defining slice of pointers. Change seems negligible, and would favour consistency with other slice definitions in the model. Let me know if there is some aspect I am missing.